src/view/components/AddEntry/index.tsx

```diff
@@ -75,6 +76,20 @@ class AddEntry extends React.Component<PropsType, OwnStateType> {
 
   /** Handle pressing save button */
   private onSavePress = () => {
+    // Make sure that the fields don't contain negative values
+    if (this.state.amountOfProduct < 0) {
+      Alert.alert('Error', 'Amount must a positive number');
+      return;
+    }
+    if (parseInt(this.state.quality, radix) < 0) {
+      Alert.alert('Error', 'Quality must a positive number');
+      return;
+    }
+    if (this.state.costPerUnit < 0) {
+      Alert.alert('Error', 'Cost per unit must a positive number');
+      return;
+    }
+
```

Comment:
This is good for the purpose of showing errors and such, but I think we shouldn't handle errors this way, user input validation should happen _once_ the user makes an error, (ie, in `onChangeAmount(newAmount: string)`) for example. You can do something like

```
  private onChangeAmount = (newAmount: string) => {
    if (parseInt(newAmount, radix) < 0) {
      // handle error
    } else {
      this.setState(state => ({ amountOfProduct: parseInt(newAmount, radix) }));
    }
  }
```

Where `handle error` can be something similar to
https://material.io/guidelines/patterns/errors.html#

This way, the user knows the mistake they made the second they make it.


Also, we should prevent them from hitting `save` until they input validated data. We can do soemthing like 
```
<Button block info={isInfo} primary={isPrimary} onPress={onPress} disabled={!allValid}>
    <Text>{text}</Text>
</Button>
```

Reply:
Check AddFarmer in `moath/AT-186/inputy-validation`

Reply:
@nscassar I added support for this in #188 since we need this tonight. You should still look into this for future reference.

Reply:
Sorry, I was just able to look at this now. I agree that your method is far superior. Since I was unable to test if my method even worked, I simply pushed what I had, hoping that it would suffice. Thank you, however, for changing it properly, and I will look into your how your solution works